OpenRA.Mods.RA/Traits/GpsWatcher.cs

```diff
@@ -58,38 +60,38 @@ public void Launch(Actor source, GpsPowerInfo info)
 		{
 			source.World.Add(new DelayedAction(info.RevealDelay * 25, () =>
 			{
-				Launched = true;
+				enabled = true;
 				Refresh();
 			}));
 		}
 
 		void Refresh()
 		{
-			RefreshGranted();
+			RefreshInner();
 
-			foreach (var i in owner.World.ActorsWithTrait<GpsWatcher>())
-				i.Trait.RefreshGranted();
+			// Refresh the state of all allied players (including ourselves, again...)
+			foreach (var i in owner.World.ActorsWithTrait<GpsWatcher>().Where(kv => kv.Actor.Owner.IsAlliedWith(owner)))
+				i.Trait.RefreshInner();
 		}
 
-		void RefreshGranted()
+		void RefreshInner()
 		{
-			var wasGranted = Granted;
-			var wasGrantedAllies = GrantedAllies;
+			var wasActive = Active;
 
-			Granted = sources.Count > 0 && Launched;
-			GrantedAllies = owner.World.ActorsHavingTrait<GpsWatcher>(g => g.Granted).Any(p => p.Owner.IsAlliedWith(owner));
+			granted = sources.Count > 0 && enabled;
+			grantedAllies = owner.World.ActorsHavingTrait<GpsWatcher>(g => g.granted).Any(p => p.Owner.IsAlliedWith(owner));
```

Comment:
This doesn't affect anything in practice, but should indeed be fixed as part of a larger cleanup.